### ISSUE TITLE
[Bug Report]  [el-image] Image resources will be requested twice

### ISSUE THREAD

**Issue by sleepq123** [element-plus/element-plus#4683]
### Element Plus version
1.2.0-beta.5

### OS/Browsers version
macos / chrome  96.0.4664.55

### Vue version
3.2.23

### Reproduction Link
https://codepen.io/sleepq123/pen/xxXZRea

### Steps to reproduce
1. Refresh browser
2. View picture request

### What is Expected?
Picture resource request once

### What is actually happening?
Picture resource request twice

**Comment by sleepq123** [comment]
![image](https://user-images.githubusercontent.com/31381291/144800750-a39eaf8e-5dd0-4708-adce-91fda340ff8c.png)

**Comment by sxzz** [comment]
Please try [link](https://play.element-plus.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcyc7XG5pbXBvcnQgeyBFbElucHV0IH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5zZXR1cEVsZW1lbnRQbHVzKCk7XG5jb25zdCBpbWcgPSByZWYoJycpXG4gc2V0VGltZW91dCgoKSAgPT4ge1xuICAgaW1nLnZhbHVlID0gXG4gICAgICdodHRwczovL2ltZy5hbGljZG4uY29tL2ltZ2V4dHJhL2kzLzk0NTA0NTk1L08xQ04wMWwyclR6NTFqb1lkd1VBak5PLTk0NTA0NTk1LmpwZz8xPTInXG4gfSwgMTAwMClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG5cdDxlbC1pbWFnZSA6c3JjPVwiaW1nXCIgc3R5bGU9XCJ3aWR0aDogMTAwcHhcIiAvPlxuPC90ZW1wbGF0ZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XCJpbXBvcnRzXCI6e319In0=).

Infact, I can't reproduce it.

**Comment by cokemine** [comment]
@sleepq123 I can not reproduce it. The PR change `v-if` to `v-show` which will render the img tag even if the source is wrong. And I don’t think it will solve the problem.
BTW, it is not normal for a browser to request the same image multiple times on the same page because of the cache.
You can try 
```
  <body>
    <img src="https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2" alt="" />
    <img src="https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2" alt="" />
    <script>
      const img = new Image();
      img.src = 'https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2';
      setTimeout(() => {
        const img = new Image();
        img.src = 'https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2';
      }, 1000);
    </script>
  </body>
```
The browser only request the same picture once, not four times

**Comment by sleepq123** [comment]
@cokemine You're right, When I find this problem and look at the source code, I guess the possible cause:

1. `new Image()` Will initiate a request
2. `IMG tag` also Will initiate a request

I tried to explain why the number of requests was normal after it was changed to v-show.

Before viewing the network, you need check disable cache link this.
![image](https://user-images.githubusercontent.com/31381291/145927736-cdcf4c25-cf49-4dd7-8f94-ba1430d06447.png)

#### The following is some examples to simulate v-if and v-show: ####
> Normal condition example（request once）

```html
<body>
  <img
    src="https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2"
  />
  <script>
    const img = new Image();
    img.src =
      "https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2";
  </script>
</body>
```

> Analog v-if example（request twice）

```html
<body>
  <div id="app"></div>
  <script>
    const img = new Image();
    img.addEventListener("load", (e) => {
      setTimeout(() => {
        let app = document.getElementById("app");
        var image = document.createElement("img");
        image.src =
          "https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2";
        app.appendChild(image);
      }, 100);
    });
    img.src =
      "https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2";
  </script>
</body>
```

>  Analog v-show example（only request once） 

```html
<body>
  <img
    id="image"
    style="display: none;"
    src="https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2"
  />
  <script>
    const img = new Image();
    img.addEventListener("load", (e) => {
      setTimeout(() => {
        let image = document.getElementById("image");
        image.style.display = "";
      }, 100);
    });
    img.src =
      "https://img.alicdn.com/imgextra/i3/94504595/O1CN01l2rTz51joYdwUAjNO-94504595.jpg?1=2";
  </script>
</body>
```

- `v-show="false"` is equivalent to `display: none`, but it will also load image resources.

- `v-if="false"` is equivalent to `removing from the DOM element` , it without loading the image resources. When you insert the DOM again,  it will reload the image resources.

- The Image() constructor creates a new HTMLImageElement instance. It is functionally equivalent to document.createElement('img').

- **In the same rendering, the browser will reuse the picture resource cache of the same address. But last image cache will not be read at the next redraw.** (I guess🙈)

Therefore, The purpose of changing `v-if` to `v-show` is to put `HTMLImageElement'referring to the same picture address into once DOM tree rendering, so that the browser can reuse image resources and reduce repeated requests.

**Comment by sleepq123** [comment]
> Please try [link](https://play.element-plus.org/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcyc7XG5pbXBvcnQgeyBFbElucHV0IH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5zZXR1cEVsZW1lbnRQbHVzKCk7XG5jb25zdCBpbWcgPSByZWYoJycpXG4gc2V0VGltZW91dCgoKSAgPT4ge1xuICAgaW1nLnZhbHVlID0gXG4gICAgICdodHRwczovL2ltZy5hbGljZG4uY29tL2ltZ2V4dHJhL2kzLzk0NTA0NTk1L08xQ04wMWwyclR6NTFqb1lkd1VBak5PLTk0NTA0NTk1LmpwZz8xPTInXG4gfSwgMTAwMClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG5cdDxlbC1pbWFnZSA6c3JjPVwiaW1nXCIgc3R5bGU9XCJ3aWR0aDogMTAwcHhcIiAvPlxuPC90ZW1wbGF0ZT4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XCJpbXBvcnRzXCI6e319In0=).
> 
> Infact, I can't reproduce it.

you can try check disable cache link this. It will reproduce.
![image](https://user-images.githubusercontent.com/31381291/145928499-9abe0910-431b-4c4b-bded-f97df9c000c8.png)

**Comment by cokemine** [comment]
@sleepq123 But why user will enable "Disabled cache" in Developer Tools. Reusing cache is the default behavior of browsers. Turning on the "Disabled cache" option by default is not normal.

**Comment by Dsaquel** [comment]
Works on newer versions, even with 'disable cache'.
<img width="1336" height="645" alt="Image" src="https://github.com/user-attachments/assets/9c993f6e-867e-416f-9be1-13810e73a2df" />
